brainiak/matnormal/covs.py

```diff
@@ -3,438 +3,475 @@
 
+        super(CovAR1, self).__init__(size)
+
         # Similar to BRSA trick I think
         if scan_onsets is None:
             self.run_sizes = [size]
-            self.offdiag_template = tf.constant(scipy.linalg.toeplitz(np.r_[0,
-                                                1, np.zeros(size-2)]),
-                                                dtype=tf.float64)
-            self.diag_template = tf.constant(np.diag(np.r_[0,
-                                                           np.ones(size-2),
-                                                           0]))
+            self.offdiag_template = tf.constant(
+                scipy.linalg.toeplitz(np.r_[0, 1, np.zeros(size - 2)]),
+                dtype=tf.float64
+            )
+            self.diag_template = tf.constant(np.diag(
+                  np.r_[0, np.ones(size - 2), 0]))
         else:
             self.run_sizes = np.ediff1d(np.r_[scan_onsets, size])
-            sub_offdiags = [scipy.linalg.toeplitz(np.r_[0, 1, np.zeros(r-2)])
-                            for r in self.run_sizes]
-            self.offdiag_template = tf.constant(scipy.sparse.
-                                                block_diag(sub_offdiags)
-                                                .toarray())
-            subdiags = [np.diag(np.r_[0, np.ones(r-2), 0])
+            sub_offdiags = [
+                scipy.linalg.toeplitz(np.r_[0, 1, np.zeros(r - 2)])
+                for r in self.run_sizes
+            ]
+            self.offdiag_template = tf.constant(
+                scipy.sparse.block_diag(sub_offdiags).toarray()
+            )
+            subdiags = [np.diag(np.r_[0, np.ones(r - 2), 0])
                         for r in self.run_sizes]
-            self.diag_template = tf.constant(scipy.sparse.
-                                             block_diag(subdiags)
-                                             .toarray())
+            self.diag_template = tf.constant(
+                scipy.sparse.block_diag(subdiags).toarray()
+            )
 
         self._identity_mat = tf.constant(np.eye(size))
 
         if sigma is None:
-            self.log_sigma = tf.Variable(tf.random_normal([1],
-                                         dtype=tf.float64), name="sigma")
+            self.log_sigma = tf.Variable(
+                tf.random.normal([1], dtype=tf.float64), name="sigma"
+            )
         else:
             self.log_sigma = tf.Variable(np.log(sigma), name="sigma")
 
         if rho is None:
-            self.rho_unc = tf.Variable(tf.random_normal([1], dtype=tf.float64),
-                                       name="rho")
+            self.rho_unc = tf.Variable(
+                tf.random.normal([1], dtype=tf.float64), name="rho"
+            )
         else:
-            self.rho_unc = tf.Variable(np.log(rho), name="rho")
+            self.rho_unc = tf.Variable(
+                2 * tf.sigmoid(self.rho_unc) - 1, name="rho")
```

Comment:
I think self.rho_unc is not defined before this line, so it should not be an argument, right?
I wonder if you were thinking of tf.Variable(scipy.special.logit(rho / 2 + 0.5), name='rho_unc')

Reply:
Yep, good catch.